src/Symfony/Component/Console/Application.php

```diff
@@ -226,7 +228,18 @@ public function doRun(InputInterface $input, OutputInterface $output)
                 $e = $event->getError();
             }
 
-            throw $e;
+            if (!($e instanceof CommandNotFoundException && !$e instanceof NamespaceNotFoundException) || 1 !== count($alternatives = $e->getAlternatives()) || !$input->isInteractive()) {
```

Comment:
Doing this here keeps dispatching a console.error event, not sure it's desired in case the user choose to run the single alternative? Especially given it's currently not taking into account the potential changes made on the event (see comment below)